### PR TITLE
Cy 3099 migrate jenkinsfile

### PR DESCRIPTION
**Pull request by yuvlar**
Migrating cloudify-agent from CircleCI to new Jenkins.
What's left to do after the PR is to set the 'master' branch to trigger on a nightly basis

### CODE REVIEW

File: jenkins/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    [envVar: 'OS_IP', vaultKey: 'ip'],
+    [envVar: 'OS_PASSWORD', vaultKey: 'password']]],
+]
+
+@Library('pipeline-shared-library') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label "agent-${env.BUILD_NUMBER}"
```
Comment (warolv) [inline jenkins/Jenkinsfile:16]: let's rename it to cloudify-agent-, we have too many agents

File: jenkins/bp/ec2-py27-blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    type: string
+    default: 'eu-west-1b'
+
+  instance_type:
+    type: string
+    default: t3.medium
+
+  aws_image_name:
+    type: string
+    default: 'ami-0b850cf02cc00fdc8'
```
Comment (warolv) [inline jenkins/bp/ec2-py27-blueprint.yaml:26]: what's that ami? you created new AMI for this?

File: jenkins/bp/ec2-py27-blueprint.yaml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        ImageId: { get_input:  aws_image_name }
+        InstanceType: { get_input: instance_type }
+        kwargs:
+          Placement:
+            AvailabilityZone: { get_input:  availability_zone }
+          UserData: { get_attribute: [ cloud_init, cloud_config ] }
+          BlockDeviceMappings:
+          - DeviceName: '/dev/xvda'
+            Ebs:
+              VolumeSize: 22
```
Comment (warolv) [inline jenkins/bp/ec2-py27-blueprint.yaml:81]: Do you need a big volume? 8Gb not enough?

File: jenkins/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+      parallel {
+        stage('Configure & Run-tests py27 on EC2 instance'){
+          steps{
+            catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE'){
+              container('py27'){
+                dir("${env.WORKSPACE}/${env.PROJECT}/jenkins"){
+                  withVault([configuration: configuration, vaultSecrets: secrets]){
+                    sh script:"""#!/bin/bash
+                      ssh -i ~/.ssh/ec2_ssh_key -l centos \$(cat capabilities.json | jq '.endpoint.value' | tr -d '"') /bin/bash << 'EOT'
+sudo yum install -y epel-release git vim python-virtualenv
```
Comment (warolv) [inline jenkins/Jenkinsfile:154]: I think you need to create AMI with preinstalled software

**Review by warolv — CHANGES_REQUESTED** [review]

File: jenkins/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        stage ('py3_compat'){
+          steps{
+            sh script: "mkdir -p ${env.WORKSPACE}/py3_compat && cp -rf ${env.WORKSPACE}/${env.PROJECT}/. ${env.WORKSPACE}/py3_compat", label: "copying repo to seperate workspace"
+            container('py27'){
+              dir("${env.WORKSPACE}/py3_compat"){
+                echo 'install futurize'
+                sh 'pip install future --user'
+                echo 'find python3-incompatible code'
+                sh '''#!/bin/bash
+                FUTURIZE="futurize ."
```
Comment (warolv) [inline jenkins/Jenkinsfile:45]: I see we using this 'futurize' thing a lot, maybe we can extract it to a shared library? what you think?

**Review by warolv — CHANGES_REQUESTED** [review]

File: jenkins/Jenkinsfile (hunk 5 above)
Comment (yuvalrap) [reply jenkins/Jenkinsfile:45]: That sounds like a great idea, it does come up a lot

File: jenkins/Jenkinsfile (hunk 1 above)
Comment (yuvalrap) [reply jenkins/Jenkinsfile:16]: ok, np

File: jenkins/Jenkinsfile (hunk 4 above)
Comment (yuvalrap) [reply jenkins/Jenkinsfile:154]: regarding the AMI, I'm not sure how much that will benefit the process, I prefer those software (especially the git clone and the test/dev requirements) to be up-to-date.
If you think some parts can still be preinstalled in an AMI, let me know which ones and I'll create one

File: jenkins/bp/ec2-py27-blueprint.yaml (hunk 3 above)
Comment (yuvalrap) [reply jenkins/bp/ec2-py27-blueprint.yaml:81]: I'll try to run it with 8gb, might be enough

File: jenkins/Jenkinsfile (hunk 5 above)
Comment (yuvalrap) [reply jenkins/Jenkinsfile:45]: Should I open a Jira for it?

File: jenkins/Jenkinsfile (hunk 1 above)
Comment (yuvalrap) [reply jenkins/Jenkinsfile]: done.

File: jenkins/bp/ec2-py27-blueprint.yaml (hunk 3 above)
Comment (yuvalrap) [reply jenkins/bp/ec2-py27-blueprint.yaml]: executed well on 8gb

File: jenkins/Jenkinsfile (hunk 5 above)
Comment (warolv) [reply jenkins/Jenkinsfile:45]: no, extract to a shared library, if you will open the JIRA it will not happen and we using this a lot I see

**Review by warolv — APPROVED** [review]

**Pull request merged**